public/media_engagement/imaqal_media_engagement.js

```diff
@@ -654,5 +654,25 @@ const update = (data) => {
             yLimitSentFiltered = this.value
             draw10MinSentGraph(yLimitSentFiltered)
         }
-    });                                                                                                                                                                                  
+    });                        
+    
+    var fullDateFormat = d3.timeFormat("%c")	
+
+    // Update timestamp of update and reset formatting
+    const timestamp = new Date()
+    d3.select("#lastUpdated").classed("alert", false).text(fullDateFormat(timestamp))
+
+    function setLastUpdatedAlert() {
+        // Calculate time diff bw current and timestamp
+        var currentTime = new Date()
+        var difference_ms = (currentTime.getTime() - timestamp.getTime())/60000
+        var difference_minutes = Math.floor(difference_ms % 60)
+     
+        // if updated more than 20 min ago >> reformat
+        if (difference_minutes > 20) {
+            d3.select("#lastUpdated").classed("alert", true)
+        }
+    }
+
+    setInterval(setLastUpdatedAlert, 1000)
```

Comment:
Is this going to add new callbacks every time the data updates? If so we'll see a gradual accumulation of the callbacks over time?

Reply:
Yes, good point. I tried to move the callback out of the data update function by making the function global defining it like window.setLastUpdateAlert, but then neither function was called. Any ideas?